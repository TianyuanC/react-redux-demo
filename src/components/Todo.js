import React from "react";
import { useActions } from "../effects/useActions";
import { toggleTodo } from "../redux/actions";

export default ({ todo }) => {
    const { toggleTodoAction } = useActions(
        {
            toggleTodoAction: id => toggleTodo(id)
        },
        []
    );
    const contentCss =
        todo && todo.completed
            ? "todo-item__text todo-item__text--completed"
            : "todo-item__text";
    return (
        <li className="todo-item" onClick={() => toggleTodoAction(todo.id)}>
            {todo && todo.completed ? "ðŸ‘Œ" : "ðŸ‘‹"}{" "}
            <span className={contentCss}>{todo.content}</span>
        </li>
    );
};
