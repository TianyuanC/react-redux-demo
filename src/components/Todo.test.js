import React from "react";
import Todo from "./Todo";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../redux/store";

describe("Todo", () => {
    it("completed item", () => {
        const component = renderer.create(
            <Provider store={store}>
                <Todo todo={{ content: "test", completed: true }} />
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("incompleted item", () => {
        const component = renderer.create(
            <Provider store={store}>
                <Todo todo={{ content: "test2", completed: false }} />
            </Provider>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
