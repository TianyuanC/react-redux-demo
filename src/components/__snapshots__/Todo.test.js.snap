// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo completed item 1`] = `
<li
  className="todo-item"
  onClick={[Function]}
>
  ðŸ‘Œ
   
  <span
    className="todo-item__text todo-item__text--completed"
  >
    test
  </span>
</li>
`;

exports[`Todo incompleted item 1`] = `
<li
  className="todo-item"
  onClick={[Function]}
>
  ðŸ‘‹
   
  <span
    className="todo-item__text"
  >
    test2
  </span>
</li>
`;
